### PR TITLE
Support for SHA1 visual passwords on version 12 server

### PR DESCRIPTION
**Pull request by 4jandals**
After upgrading to version 12 of Loxone server, the alarm controls stopped working. i.e. it was no longer possible to arm or disarm the alarm. The only error in the logs was that a `"Code": "500"` response which according to the logs indicates the password was incorrect.

After debugging, identified that the server indicated the visual password was hashed using SHA1, but the logic in api.py was hardcoded to use SHA256 if version >= 12.0

After more investigation and reading the Loxone API docs, I think that the SHA is defined by server version that created the user record, so a version 12 server can contain both SHA1 and SHA256 user records. 

Possibly reseting the password would be a workaround, but it's easy enough to recognise the `hashAlg` returned by the Loxone server and apply it appropriately.

I've tested with both a SHA1 and SHA256 user and it's now working for me with these changes

### CODE REVIEW

File: custom_components/loxone/api.py [hunk 1]
```diff
@@ -839,10 +852,10 @@ def read_user_salt_responce(self, reponse):
         value = js['LL']['value']
         self.key = value['key']
         self.salt = value['salt']
-        self.hash_alg = value.get("hashAlg", "SHA256")
+        self.hash_alg = value.get("hashAlg", "SHA1")
```
Comment (4jandals) [inline custom_components/loxone/api.py:855]: Changed the default here to SHA1 on basis that if the server doesn't return a value for `hashAlg` it must be older than v11

**Comment by JoDehli** [comment]
Thanks for your support!!

**Comment by JoDehli** [comment]
@4jandals I work together with another developer on a seperate api for communicating with the loxone. There we already changed the way the hashAlg is used. When we have the first release maybe you can test it on your machine. It is on the https://github.com/JoDehli/PyLoxone/tree/api_separation branch. And the reo for the api is https://github.com/JoDehli/pyloxone-api

**Comment by JoDehli** [comment]
@4jandals I made a discord server which I want so bundle all sort of ideas and problems. Maybe you want to join. 

I already made a pip package for the pyloxone-api and updated the api-separation branch. So if you want to test it if all of your changes you made are included you can checkout this branch. It will download the api and uses this one. The big goal is it to get it in the main branch from home-assistant. But until this happens it is a long way :-)

**Pull request merged**